### PR TITLE
Various bug fixes and tests (4-2-stable) 

### PR DESCRIPTION
**Pull request by alanking**
Addresses https://github.com/irods/irods/issues/6068
Addresses https://github.com/irods/irods/issues/6100
Addresses https://github.com/irods/irods/issues/6285
Addresses https://github.com/irods/irods/issues/6365
Addresses https://github.com/irods/irods/issues/6428
Addresses https://github.com/irods/irods/issues/6838

Cherry-picked and adapted from https://github.com/irods/irods/pull/6836

Note the base commit which was cherry-picked from main so that I could build this.

Running core tests again

### CODE REVIEW

File: scripts/irods/lib.py [hunk 1]
```diff
@@ -757,9 +757,21 @@ def get_replica_size(session, data_name, replica_number):
...
+def replica_exists(session, logical_path, replica_number):
     out = session.run_icommand(['iquest',
-        "select DATA_ID where DATA_NAME = '{}' and DATA_REPL_NUM = '{}'"
-        .format(data_name, str(replica_number))])[0]
+        "select DATA_ID where COLL_NAME = '{}' and DATA_NAME = '{}' and DATA_REPL_NUM = '{}'"
+        .format(os.path.dirname(logical_path),
+                os.path.basename(logical_path),
+                str(replica_number))])[0]
+
+    return 'CAT_NO_ROWS_FOUND' not in out
```
[the remark this answers is not in the document]
Comment (alanking) [reply scripts/irods/lib.py:767]: Hmm... probably. What should we do if the error code is not 0? Does `CAT_NO_ROWS_FOUND` cause the error code to be non-0?

[the remark this answers is not in the document]
Comment (alanking) [reply scripts/irods/lib.py:767]: To answer my own question, `CAT_NO_ROWS_FOUND` does return a non-zero code.

It's still not clear to me what we want to do in the other cases, though. Seeing as I just lifted these two functions from main, I have filed a separate issue to address this for all branches: https://github.com/irods/irods/issues/6841

[the remark this answers is not in the document]
Comment (korydraughn) [reply scripts/irods/lib.py:767]: Yeah, don't get too hung up on that. We can address it later.

[the remark this answers is not in the document]
Comment (alanking) [reply scripts/irods/lib.py:767]: Okey dokey. I will resolve this, then.

**Comment by alanking** [comment]
Got two test failures.

One had to do with the missing `lib.py` function which the latest commit addresses.

The other was in `test_rule_engine_plugin_framework`, where an expected `ils` output changed from finding an error code to the name of the error. A similar change was made in 0e36b87dde208208304c0e81ba06e8901901ff02 due to `irods::experimental::filesystem` errors getting more specific. I see that the linked issue is targeting 4.2.12. @korydraughn - Should I just leave this alone and wait for that inevitable cherry-pick, or go ahead and fix it now so that the tests pass for me? I'm fine with either way.

**Comment by korydraughn** [comment]
If it's easy for you to tweak the test, then go for it. I'll adapt the cherry-picks for that.

**Comment by alanking** [comment]
Test has been tweaked. Confirmed that federation tests are now passing (again) and I'm running the core tests once more to make sure everything still behaves.

I'm pretty sure topology tests are not passing for any branch right now, but I will address that in a separate PR as it is unrelated to these changes.

**Comment by alanking** [comment]
Now also addresses #6842. Running tests again for good measure since there are some real changes in there...

**Comment by alanking** [comment]
Observed a failure in `test_irm` from the new test. It seems that the `itouch` and `irm` can happen in the same second, which causes the collection mtime update check to fail (given the seconds granularity). I will need to introduce a `time.sleep` to ensure that the timestamps will be different. I confirmed that the test passes consistently with such a change. This change will need to be ported to main and 4-3-stable as well.

**Comment by alanking** [comment]
Last commit causes test to pass consistently, so all tests are passing now. Any other thoughts?

**Comment by trel** [comment]
third commit message... for 6285

`does not have permissions for gets
updated even if the data object`

having trouble parsing that

also use of 'bonkers' is fun.

**Comment by trel** [comment]
commit message for 6428

in tact when modifying the collection -> "intact"

**Comment by alanking** [comment]
2 points on those messages...

1. Both commit messages now exist in main and 4-3-stable. Happy to change it, but a divergence would occur.
2. The sentence does make sense, but perhaps is poorly structured and/or is missing a comma somewhere lol

> This adds a test that ensures that irepl -a being run as an admin user
> on a data object for which the admin user has no permissions gets
> replicated even if the data object has a bonkers name and exists inside of
> a bonkers collection and is a zero-length replica.

**Comment by trel** [comment]
ah i see it now.  broke my parser.

keep the messages the same - all good.

**Review by korydraughn — DISMISSED** [review]
Looks like everything is in order.

Let's get it merged.

**Comment by alanking** [comment]
#'d

**Review by korydraughn — APPROVED** [review]
Merge when ready.

**Pull request merged**